compiler/AST/build.cpp

```diff
@@ -1736,6 +1719,14 @@ buildClassDefExpr(const char* name,
                   Flag        isExtern,
                   const char* docs) {
   AggregateType* ct = toAggregateType(type);
+  if (strcmp("string", name) == 0) {
+    *dtString = *ct;
+    dtString->fields.parent = dtString;
+    dtString->inherits.parent = dtString;
```

Comment:
It is not obvious to me why you need to set fields.parent and inherits.parent here.